src/Momento.Sdk/Responses/Vector/IndexInfo.cs

```diff
@@ -10,30 +12,52 @@ public class IndexInfo
     /// </summary>
     public string Name { get; }
 
+    /// <summary>
+    /// The number of dimensions in the index.
+    /// </summary>
+    public int NumDimensions { get; }
+
+    /// <summary>
+    /// The similarity metric used by the index.
+    /// </summary>
+    public SimilarityMetric SimilarityMetric { get; }
+
     /// <summary>
     /// Constructs an IndexInfo.
     /// </summary>
     /// <param name="name">The name of the index.</param>
-    public IndexInfo(string name)
+    /// <param name="numDimensions">The number of dimensions in the index.</param>
+    /// <param name="similarityMetric">The similarity metric used by the index.</param>
+    public IndexInfo(string name, int numDimensions, SimilarityMetric similarityMetric)
     {
         Name = name;
+        NumDimensions = numDimensions;
+        SimilarityMetric = similarityMetric;
     }
 
+
     /// <inheritdoc />
     public override bool Equals(object obj)
     {
-        return obj is IndexInfo other && Name == other.Name;
+        return obj is IndexInfo other && Name == other.Name && NumDimensions == other.NumDimensions && SimilarityMetric == other.SimilarityMetric;
     }
 
     /// <inheritdoc />
     public override int GetHashCode()
     {
-        return Name.GetHashCode();
+        unchecked
+        {
+            var hash = 17;
+            hash = hash * 23 + Name.GetHashCode();
+            hash = hash * 23 + NumDimensions.GetHashCode();
+            hash = hash * 23 + SimilarityMetric.GetHashCode();
```

Comment:
JC where do 17 and 23 come from? Could we make named constants for them for what they represent? Just not clear to me looking at code why they equal to what they are.

Reply:
These are prime numbers used to combine the hash codes. There is a .NET standard implementation to combine the hash code representations, but we are targeting such old frameworks we don't have access to it, hence have to copy that here. See here https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/HashCode.cs#L58

Reply:
I see ok maybe just comment in code linking out to this at least? I would never have figured this out on my own or had context why were doing.